app/src/main/java/io/appium/uiautomator2/handler/ScrollTo.java

```diff
@@ -24,20 +28,54 @@ public ScrollTo(String mappedUri) {
     }
 
     @Override
-    protected AppiumResponse safeHandle(IHttpRequest request) throws JSONException,
-            UiObjectNotFoundException {
+    protected AppiumResponse safeHandle(IHttpRequest request)
+        throws JSONException, UiObjectNotFoundException
+    {
         String json = getPayload(request).toString();
-        String selector = "$.params.selector", uiSelectorString, scrollToString = "";
-        uiSelectorString = JsonPath.compile(selector).read(json);
-        Device.waitForIdle();
-        // Extracting (\"Radio Group\") text from the String
-        // TODO This logic needs to be changed according to the request body from the Driver
-        Matcher m = Pattern.compile("\\(\"([^)]+)\"\\)").matcher(uiSelectorString);
-        while (m.find()) {
-            scrollToString = m.group(1);
+        String strategy  = "$.params.strategy";
+        String selector  = "$.params.selector";
+        String maxSwipes = "$.params.maxSwipes";
+
+        String strategyString = JsonPath.compile(strategy).read(json);
+        String selectorString = JsonPath.compile(selector).read(json);
+
+        int maxSearchSwipes;
+        try {
+            maxSearchSwipes = JsonPath.compile(maxSwipes).read(json);
+        } catch (InvalidPathException e) {
+            Logger.info("Parameter 'maxSwipes' not specified. Using default value.");
```

Comment:
The parameter ... is either not specified or mistyped. Using 0 as the default value.

Reply:
Log message wording has been changed as suggested.